src/nacl/hash.py

```diff
@@ -17,10 +17,80 @@
 import nacl.bindings
 import nacl.encoding
 
+BYTES = nacl.bindings.crypto_generichash_BYTES
+BYTES_MIN = nacl.bindings.crypto_generichash_BYTES_MIN
+BYTES_MAX = nacl.bindings.crypto_generichash_BYTES_MAX
+KEYBYTES = nacl.bindings.crypto_generichash_KEYBYTES
+KEYBYTES_MIN = nacl.bindings.crypto_generichash_KEYBYTES_MIN
+KEYBYTES_MAX = nacl.bindings.crypto_generichash_KEYBYTES_MAX
+SALTBYTES = nacl.bindings.crypto_generichash_SALTBYTES
+PERSONALBYTES = nacl.bindings.crypto_generichash_PERSONALBYTES
+
+_b2b_hash = nacl.bindings.crypto_generichash_blake2b_salt_personal
+
 
 def sha256(message, encoder=nacl.encoding.HexEncoder):
     return encoder.encode(nacl.bindings.crypto_hash_sha256(message))
 
 
 def sha512(message, encoder=nacl.encoding.HexEncoder):
     return encoder.encode(nacl.bindings.crypto_hash_sha512(message))
+
+
+def generichash(data, digest_size=BYTES, key=b'',
+                encoder=nacl.encoding.HexEncoder):
+    """
+    One-shot generichash digest
+
+    :param data:
+    :type data: bytes
+    :param digest_size: the requested digest size; must be between
+                        :py:data:`.BYTES_MIN` and :py:data:`.BYTES_MAX`;
+                        the default length is :py:data:`.BYTES`
+    :type digest_size: int
+    :param key: the key to be set for keyed MAC/PRF usage; if set, key size
+                should be comprised between :py:data:`.KEYBYTES_MIN`
+                and :py:data:`.KEYBYTES_MAX`
+    :type key: bytes
+    :param encoder: the encoder to use on returned digest
+    :type encoder: :py:data:`types.FunctionType`
+    :return: the encoded digest of input data
+    :rtype: the return type of `encoder`
+    """
+
+    digest = _b2b_hash(data, digest_size=digest_size, key=key)
+    return encoder.encode(digest)
+
+
+def blake2b(data, digest_size=BYTES, key=b'',
+            salt=b'', person=b'',
+            encoder=nacl.encoding.HexEncoder):
+    """
+    One-shot blake2b digest
+
+    :param data: the digest input byte sequence
+    :type data: bytes
+    :param digest_size: the requested digest size; must be between
+                        :py:data:`.BYTES_MIN` and :py:data:`.BYTES_MAX`;
+                        the default length is :py:data:`.BYTES`
+    :type digest_size: int
+    :param key: the key to be set for keyed MAC/PRF usage; if set, key size
+                should be comprised between :py:data:`.KEYBYTES_MIN`
+                and :py:data:`.KEYBYTES_MAX`
+    :type key: bytes
+    :param salt: an initialization salt of lenght :py:data:`.SALTBYTES`.
+                 It will be zero-padded or truncated if needed.
+    :type salt: bytes
+    :param person: a personalization string of lenght
```

Comment:
Any thoughts about calling this `personalization=` instead of `person=`? The latter sounds pretty weird.